source/management_console/index.js

```diff
@@ -49,10 +137,57 @@ var isAllowed = function (path) {
   return false;
 };
 
+app.post('/login', function(req, res) {
+  req.session.serviceid = req.body.id;
+  req.session.servicekey = req.body.key;
+  res.status(200).send('Successfully logged in');
+});
+
+app.get('/login', function(req, res) {
+    if (req.session.serviceid) {
+      let header = req.headers.authorization;
+      const timestamp = new Date().getTime();
+      const cnounce = Math.floor(Math.random() * 99999);
+      const toSign = timestamp + ',' + cnounce;
+      header += ',mauth_serviceid=';
+      header += req.session.serviceid;
+      header += ',mauth_cnonce=';
+      header += cnounce;
+      header += ',mauth_timestamp=';
+      header += timestamp;
+      header += ',mauth_signature=';
+      header += calculateSignature(toSign, req.session.servicekey);
+      req.headers.authorization = header;
+      req.url = '/services/' + req.session.serviceid;
+      proxy.web(req, res, {target: apiHost});
```

Comment:
I think you don't need this "if" part. You proxy the "login" request to REST API server?